### PR TITLE
MaintenanceManagementService  improvement - Allow implementation of OperationInterface to return null

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#2034 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Maintenance Management service fails silently when user provided OperationInterface returns null in take/freeInstance API.
We should allow user provided OperationInterface to return null.

### Tests

- [X] The following tests are written for this issue:

Modified TestOperationImpl.java.
TestInstanceAccessor will cover null case.

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java
Comment (qqu0127) [inline helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java:91]: Is `status` a class attribute? Let's follow the naming convention like `_status`, thanks.

**Comment by qqu0127** [comment]
The impl looks good with minor comments, but here are some general questions on the design.
Could you share more context on why the service silently fail? 
Also I'm wondering if a null value makes sense in the first place. Will something like a default or empty value be better?

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java
Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java]: Good point. Updated. 
I think if we get null `info`, we could keep status the same..?

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java
Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java]: TFTR. Updated.

**Comment by xyuanlu** [comment]
> The impl looks good with minor comments, but here are some general questions on the design. Could you share more context on why the service silently fail? Also I'm wondering if a null value makes sense in the first place. Will something like a default or empty value be better?

Thanks for the reply.
If we get null for MaintenanceManagementInstanceInfo, then in MaintenanceManagementService.takeFreeSingleInstanceHelper() will throw npr exception causing operation failure.

This function execute user implemented interface `OperationInterface`. It  should make sense to allow other user to return null.

**Review by qqu0127 — APPROVED** [review]
LGTM, please address other comments and get committer approval.

**Review by desaikomal — COMMENTED** [review]
LGTM ... simple change..

**Comment by xyuanlu** [comment]
This PR is ready to be merged. Approved by @qqu0127 @desaikomal //Thanks!
Commit message:
Allow implementation of OperationInterface to return null

**Pull request merged**
